File: src/main/scala/HelperUtils/Cost.java
package HelperUtils;

import org.cloudbus.cloudsim.cloudlets.Cloudlet;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.List;

public class Cost {

    private static final Logger logger = LoggerFactory.getLogger(Cost.class.getSimpleName());

    public static float executionCost(List<Cloudlet> cloudlets) {
        float cost = 0.0f;
        for (Cloudlet cl : cloudlets) {
            cost += cl.getTotalCost();
            logger.info("Cost to execute cloudlet {} on datacenter {} = {}"
                    , cl.getId(), cl.getLastTriedDatacenter().getId(), (float) cl.getTotalCost());
        }
        return cost;
    }
}

File: src/main/scala/HelperUtils/Cost.scala
package HelperUtils

import Simulations.SaasSim
import org.cloudbus.cloudsim.brokers.DatacenterBroker
import org.cloudbus.cloudsim.cloudlets.Cloudlet
import org.cloudbus.cloudsim.vms.VmCost

import java.util
import java.util.Comparator
import scala.jdk.CollectionConverters.*
import scala.jdk.javaapi.CollectionConverters.asJava

class Cost

object Cost:

  val logger = CreateLogger(classOf[Cost])

  /**
   * Calculates the cost of an execution, based on the Datacenters cost configurations.
   *
   * @param boker          DatacenterBroker object.

   * @return Total cost of execution of the Simulation run.
   */
    def findCost(broker: DatacenterBroker) : Double = {

      var totalCost: Double = 0.0

      for (cl <- broker.getVmCreatedList.asScala) {
        val cost: VmCost = new VmCost(cl)

        logger.info("Working on datacenter {}, {}", cl.getLastTriedDatacenter.getId, cost)
        totalCost += cost.getTotalCost

      }
      totalCost

    }

